pkg/dashboard/server/flux.go

```diff
@@ -101,3 +114,44 @@ func parseRev(rev string) (string, error) {
 
 	return parts[1], nil
 }
+
+func updateGitopsCommitStatuses(
+	gitopsRepoCache *nativeGit.RepoCache,
+	perf *prometheus.HistogramVec,
+	store *store.Store,
+	repoName, env, eventMessage string,
+	repoPerEnv bool,
+) error {
+	since := time.Now().Add(-24 * time.Hour)
+	repo, pathToCleanUp, err := gitopsRepoCache.InstanceForWrite(repoName) // using a copy of the repo to avoid concurrent map writes error
+	defer gitopsRepoCache.CleanupWrittenRepo(pathToCleanUp)
+	if err != nil {
+		return err
+	}
+
+	releases, err := gitops.Releases(repo, "", env, repoPerEnv, &since, nil, 10, "", perf)
```

Comment:
And then you ask for 10 commits. What if 12 was made since the last apply?

Make the code handle any number of commits. Don't rely on assumptions.